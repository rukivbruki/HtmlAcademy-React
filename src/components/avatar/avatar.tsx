import * as React from 'react';
import {Link, withRouter} from "react-router-dom";
import {connect} from "react-redux";
import Paths from '../../router/paths';
import {UserType, HistoryType} from '../../types.js';
import {getUser, getIsAuth} from "../../reducer/user/selectors";

const ORIGIN = `https://htmlacademy-react-3.appspot.com`;

const defautUsetData = {
  id: 1,
  name: ``,
  email: ``,
  avatarUrl: ``
};

interface Props {
  history?: HistoryType;
  isAuth?: boolean;
  userData?: UserType;
}

const Avatar = ({history, isAuth, userData = defautUsetData}: Props) => {
  const {avatarUrl = ``} = userData;

  const handleClickAvatar = () => {
    history.push(Paths.MY_LIST);
  };
  if (isAuth) {
    return (
      <div className="user-block__avatar" onClick={handleClickAvatar}>
        <img
          src={`${ORIGIN}${avatarUrl}`}
          alt="User avatar"
          width="63"
          height="63"
        />
      </div>
    );
  } else {
    return (
      <Link to={Paths.LOGIN} className="sign-in__link">
        Sign In
      </Link>
    );
  }
};

const mapStateToProps = (state) => ({
  userData: getUser(state),
  isAuth: getIsAuth(state)
});

export {Avatar};
export default connect(mapStateToProps)(React.memo(withRouter(Avatar)));
