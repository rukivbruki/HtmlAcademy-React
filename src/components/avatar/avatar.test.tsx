import * as React from "react";
import * as renderer from "react-test-renderer";
import {Provider} from "react-redux";
import {BrowserRouter} from "react-router-dom";
import configureStore from "redux-mock-store";
import NameSpace from "../../reducer/name-space.js";
import {Avatar} from "./avatar";

const mockStore = configureStore([]);

it(`Avatar renders correctly`, () => {
  const store = mockStore({
    [NameSpace.USER]: {
      authorizationStatus: `AUTH`,
      user: {
        id: 1,
        email: `test@email.com`,
        name: `test user`,
        avatarUrl: `avatar`
      }
    },
  });
  const tree = renderer
    .create(
        <Provider store={store}>
          <BrowserRouter>
            <Avatar />
          </BrowserRouter>
        </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
