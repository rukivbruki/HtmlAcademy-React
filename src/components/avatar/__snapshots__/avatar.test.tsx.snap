// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar renders correctly 1`] = `
<a
  className="sign-in__link"
  href="/login"
  onClick={[Function]}
>
  Sign In
</a>
`;
